skfda/inference/anova/anova_oneway.py

```diff
@@ -174,29 +175,41 @@ def _anova_bootstrap(fd_grouped, n_reps, random_state=None, p=2):
 
     sizes = [fd.n_samples for fd in fd_grouped]  # List with sizes of each group
 
-    # Estimating covariances for each group
-    k_est = [fd.cov().data_matrix[0, ..., 0] for fd in fd_grouped]
-
-    # Number of sample points for gaussian processes have to match the features
-    # of the covariances.
-    n_features = k_est[0].shape[0]
-
     # Instance a random state object in case random_state is an int
     random_state = check_random_state(random_state)
 
-    # Simulating n_reps observations for each of the n_groups gaussian processes
-    sim = [make_gaussian_process(n_reps, n_features=n_features, start=start,
-                                 stop=stop, cov=k_est[i],
-                                 random_state=random_state)
-           for i in range(n_groups)]
+    if equal_var:
+        k_est = concatenate(fd_grouped).cov().data_matrix[0, ..., 0]
```

Comment:
Convert `k_est` to a list (`k_est = [k_est] * len(fd_grouped)`) and reuse common code.